aea/aea.py

```diff
@@ -87,16 +88,19 @@ def __init__(
 
         self.max_reactions = max_reactions
         self._task_manager = TaskManager()
-        self._decision_maker = DecisionMaker(identity, wallet, ledger_apis)
+        decision_maker_handler = DecisionMakerHandler(
+            identity=identity, wallet=wallet, ledger_apis=ledger_apis
+        )
+        self._decision_maker = DecisionMaker(
+            decision_maker_handler=decision_maker_handler
+        )
```

Comment:
@marcofavorito here it would now be possible to provide a user provided DMHandler